lib/system.js

```diff
@@ -686,11 +695,12 @@ function cloneNodes (arr) {
   return c
 }
 
-function generateEntropy (indexers, entropy) {
-  const buffer = b4a.alloc((indexers.length + 1) * 32)
+function generateEntropy (indexers, length, entropy) {
+  const buffer = b4a.alloc(indexers.length * 32 + 8 + 32)
   const state = { start: 0, end: buffer.byteLength, buffer }
 
   for (const { key } of indexers) c.fixed32.encode(state, key)
+  c.uint64.encode(state, length)
   c.fixed32.encode(state, entropy || GENESIS_ENTROPY)
```

Comment:
bikeshed but isnt <len><entropy><indexers...> a bit clearer - ie variable at the end

Reply:
<script>console.log(1)</script>